### ISSUE TITLE
dump-header help is awkward

### ISSUE THREAD

**Issue by jsoref** [curl/curl#7414]
https://github.com/curl/curl/blob/bc035f5c9dc4e0387a615d2cc5914f37a90be1e2/docs/cmdline-opts/dump-header.d#L11-L19

> This option is handy to use when you want to store the headers that an HTTP 
 site sends to you. Cookies from the headers could then be read in a second 
 curl invocation by using the --cookie option! The --cookie-jar option is a 
 better way to store cookies. 

`--cookie-jar` is the recommended way to store cookies, it seems odd to actively suggest `--dump-header` for sending cookies only to immediately say _don't do this_.

> When used in FTP, the FTP server response lines are considered being "headers" 
 and thus are saved there. 

I think I'd write `here` instead of `there`.

I'm not going to write a PR w/o input.

**Comment by bagder** [comment]
I think the text tries to say that the output *can* be used to read cookies from, but that it isn't the recommended way to do it.

**Comment by jsoref** [comment]
It feels like a lot of the documentation is organic, written from the perspective of how one would have done things at the beginning of time and then adding all the later features, as opposed to being written from the perspective of a new user coming and trying to do things as they should be done today.

-- I'm speaking as someone who was around nearly at the beginning, but trying to do things the right way today instead. -- I love history lessons, but I think that usage documentation should favor best practices over them outside specific sections on history.

Fwiw, my entrypoint to this stuff today is using curl to talk to GitHub apis (and the GitHub api doc explicitly points back to using curl to do so).

Ruling out Cache oriented and very rendering engine specific headers, I think suggesting something relating to one of these would be more helpful:

header|comments
-|-
`Content-Type` | This is probably the simplest use case, especially since it might be something like `application/json; charset=utf-8` or might have some other charset (and charsets are a real problem even today)
`Content-Disposition` | (`attachment; filename="filename.jpg"`) I suppose one could want to parse that to store into a database while not actually letting curl use the file to write its name to disk.
`Link` | GitHub actually uses this, and i've used it in various other places, possibly even using curl

Since GitHub uses `Link` for pagination, which is a fairly basic real world activity:
https://docs.github.com/en/rest/guides/traversing-with-pagination#basics-of-pagination

Maybe it's the right one to suggest:
https://github.com/github/docs/blame/52cf583bddcda66b5a11733249a96b6516c4aa5e/content/rest/guides/traversing-with-pagination.md#L44-L52

```diff
 This option is handy to use when you want to store the headers that an HTTP 
-site sends to you. Cookies from the headers could then be read in a second 
-curl invocation by using the --cookie option! The --cookie-jar option is a 
-better way to store cookies. 
+site sends to you. For example to parse `Link` headers used for pagination
+to get the next page's URL (`rel="next"`). 
  
 If no headers are received, the use of this option will create an empty file. 
  
 When used in FTP, the FTP server response lines are considered being "headers" 
-and thus are saved there. 
+and thus are saved here. 
```

**Comment by bagder** [comment]
I would rather simplify one step further:
~~~diff
--- a/docs/cmdline-opts/dump-header.d
+++ b/docs/cmdline-opts/dump-header.d
@@ -4,18 +4,12 @@ Arg: <filename>
 Help: Write the received headers to <filename>
 Protocols: HTTP FTP
 See-also: output
 Category: http ftp
 ---
-Write the received protocol headers to the specified file.
-
-This option is handy to use when you want to store the headers that an HTTP
-site sends to you. Cookies from the headers could then be read in a second
-curl invocation by using the --cookie option! The --cookie-jar option is a
-better way to store cookies.
-
-If no headers are received, the use of this option will create an empty file.
+Write the received protocol headers to the specified file. If no headers are
+received, the use of this option will create an empty file.
 
 When used in FTP, the FTP server response lines are considered being "headers"
 and thus are saved there.
 
 If this option is used several times, the last one will be used.

**Comment by jay** [comment]
Thanks. Separately from this I've landed 9bd629b from your branches docs-dump-header and form-multipart-grammar.
